core/src/main/java/me/aurium/beetle/api/task/futures/SyncTaskerFuture.java

```diff
@@ -0,0 +1,338 @@
+    }
+
+    @Override
+    public <U> SyncTaskerFuture<U> thenCompose(Function<? super T, ? extends CompletionStage<U>> fn) {
+        return (SyncTaskerFuture<U>) super.thenCompose(fn);
+    }
+
+    @Override
+    public <U> SyncTaskerFuture<U> thenComposeAsync(Function<? super T, ? extends CompletionStage<U>> fn) {
+        return (SyncTaskerFuture<U>) super.thenComposeAsync(fn);
+    }
+
+    @Override
+    public <U> SyncTaskerFuture<U> thenComposeAsync(Function<? super T, ? extends CompletionStage<U>> fn, Executor executor) {
+        return (SyncTaskerFuture<U>) super.thenComposeAsync(fn,executor);
+    }
+
+    @Override
+    public <U> SyncTaskerFuture<U> handle(BiFunction<? super T, Throwable, ? extends U> fn) {
+        return (SyncTaskerFuture<U>) super.handle(fn);
+    }
+
+    @Override
+    public <U> SyncTaskerFuture<U> handleAsync(BiFunction<? super T, Throwable, ? extends U> fn) {
+        return (SyncTaskerFuture<U>) super.handleAsync(fn);
+    }
+
+    @Override
+    public <U> SyncTaskerFuture<U> handleAsync(BiFunction<? super T, Throwable, ? extends U> fn, Executor executor) {
+        return (SyncTaskerFuture<U>) super.handleAsync(fn, executor);
+    }
+
+    @Override
+    public SyncTaskerFuture<T> whenComplete(BiConsumer<? super T, ? super Throwable> action) {
+        return (SyncTaskerFuture<T>) super.whenComplete(action);
+    }
+
+    @Override
+    public SyncTaskerFuture<T> whenCompleteAsync(BiConsumer<? super T, ? super Throwable> action) {
+        return (SyncTaskerFuture<T>) super.whenCompleteAsync(action);
+    }
+
+    @Override
+    public SyncTaskerFuture<T> whenCompleteAsync(BiConsumer<? super T, ? super Throwable> action, Executor executor) {
+        return (SyncTaskerFuture<T>) super.whenCompleteAsync(action,executor);
+    }
+
+    @Override
+    public SyncTaskerFuture<T> completeAsync(Supplier<? extends T> supplier) {
+        return (SyncTaskerFuture<T>) super.completeAsync(supplier);
+    }
+
+    @Override
+    public SyncTaskerFuture<T> completeAsync(Supplier<? extends T> supplier, Executor executor) {
+        return (SyncTaskerFuture<T>) super.completeAsync(supplier, executor);
+    }
+
+    @Override
+    public SyncTaskerFuture<T> exceptionally(Function<Throwable, ? extends T> fn) {
+        return (SyncTaskerFuture<T>) super.exceptionally(fn);
+    }
+
+    @Override
+    public TaskFuture<T> toCompletableFuture() {
+        return this;
+    }
+
+    //SHIT STARTS HERE
+
+
+    @Override
+    public T get(long amount, TimeUnit unit) throws ExecutionException, InterruptedException, TimeoutException {
+        if (coordinator.isMainThread()) {
+
+            T returned;
+            if (!(returned = super.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
+                return returned;
+            }
+
+            return coordinator.waitForTimer(this,unit,amount);
+        }
+
+        if (amount <= 0L) {
+            throw new TimeoutException();
```

Comment:
Not necessary - have super.get handle this

Reply:
Does super.get already handle this?

Reply:
If you think there is a spec violation in the implementation of CompletableFuture.get, I suggest you open a bug on the JDK tracker